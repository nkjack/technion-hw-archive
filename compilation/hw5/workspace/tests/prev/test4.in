void maxTestsAssignVarToVar() {
	int x = 3;
	int y = x;
	printi(y);
	print("\n");
	printi(x);
}

void main() {
     maxTestsAssignVarToVar();
     print("\nI hate this course\n");
}
