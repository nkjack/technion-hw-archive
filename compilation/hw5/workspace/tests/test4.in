
void foo(int a, int c)
@pre(a > 0 and c > 0)
@pre(a < 10) 
{
	print("OK\n");
}

void foo2(int a, int c)
@pre(a > 0 and c > 7)
@pre(a < 10) 
{
	print("OK2\n");
}


void main() {
	foo(5, 7);
	foo2(11, 7);
}
