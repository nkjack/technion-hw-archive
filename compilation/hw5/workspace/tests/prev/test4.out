Loaded: ./exceptions.s
3
3
I hate this course
